mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/KeyRegistrationHandler.java

```diff
@@ -107,18 +109,29 @@ Registration register(MailboxListener listener, RegistrationKey key) {
         if (delivery.getBody() == null) {
             return Mono.empty();
         }
+
+        String serializedEventBusId = delivery.getProperties().getHeaders().get(EVENT_BUS_ID).toString();
+        EventBusId eventBusId = EventBusId.of(serializedEventBusId);
+
         String routingKey = delivery.getEnvelope().getRoutingKey();
         RegistrationKey registrationKey = routingKeyConverter.toRegistrationKey(routingKey);
         Event event = toEvent(delivery);
 
         return mailboxListenerRegistry.getLocalMailboxListeners(registrationKey)
+            .filter(listener -> !isLocalSynchronousListeners(eventBusId, listener))
             .flatMap(listener -> Mono.fromRunnable(Throwing.runnable(() -> listener.event(event)))
                 .doOnError(e -> LOGGER.error("Exception happens when handling event of user {}", event.getUser().asString(), e))
-                .onErrorResume(e -> Mono.empty()))
-            .subscribeOn(Schedulers.elastic())
+                .onErrorResume(e -> Mono.empty())
+                .then().cache())
+            .subscribeOn(Schedulers.parallel())
```

Comment:
Why did we change to the `parallel` one?

Don't we expect the listeners to be doing some crappy blocking stuff that would justify the elastic scheduler?